libvast/src/detail/mmapbuf.cpp

```diff
@@ -57,6 +57,8 @@ mmapbuf::mmapbuf(const path& filename, size_t size, size_t offset)
     file_size = st.st_size;
   else if (result < 0 && errno != ENOENT)
     return;
+  else
+    is_file_owner_ = true;
```

Comment:
This check auto-detects the file size. It's not intuitive why an `else` branch implies that the user owns the file. I suggest clarifying this with a comment or spliting the logic.

Reply:
> I'm not sure if I understand the implications of this PR. When and why do we want to automatically delete a file that we created? Shouldn't deletion of a file be something the user decides after having closed the streambuffer?

In general terms: Any object that allocates a resource should release that resource at the end of its lifetime. Why should that only apply to memory?
But your question prompted me to think more about the interface to `mmapbuf` and I think a better approach might be to have two sparate constructors, one for owning an the other for non-owning `mmapbuf`s. I'll append another commit to this PR for the change.

> This check auto-detects the file size. It's not intuitive why an else branch implies that the user owns the file. I suggest clarifying this with a comment or spliting the logic.

You are quite correct.